### PR TITLE
Update repos

### PR DESCRIPTION
**Pull request by aardappel**
memory64:
    https://github.com/WebAssembly/memory64/commit/3679be2c

This change was automatically generated by `update-testsuite.sh`

### CODE REVIEW

**Review by sbc100 — APPROVED** [review]

**Comment by RReverser** [comment]
@aardappel @sbc100 This repo doesn't have issues, so I'll try my luck here... Can you please update repos again? Looks like some tests are outdated.

In particular, I'm running into failures in wasmbin in tests like https://github.com/WebAssembly/testsuite/blob/01efde81028c5b0d099eb836645a2dc5e7755449/data.wast#L289-L300 which are indeed invalid and have been updated upstream since: https://github.com/WebAssembly/spec/blob/662232701afa7bef3c5b1e77362f2bab6173fca4/test/core/data.wast#L299-L310

**Comment by aardappel** [comment]
@RReverser where are you seeing these failures exactly? I though this repo was just for merging tests from proposals before they land in the official spec repo, for use in WABT etc..

Also, updating the tests in this repo is pretty easy, just run `update-tests.sh`, and it makes the new commit for you. If that doesn't work for you I'd be happy to do it.

**Comment by RReverser** [comment]
> where are you seeing these failures exactly

I'm seeing failures in my Wasm toolchain (https://github.com/GoogleChromeLabs/wasmbin) which I'm trying to upgrade to latest proposals and which depends on this repo for testing.

> Also, updating the tests in this repo is pretty easy, just run `update-tests.sh`, and it makes the new commit for you. If that doesn't work for you I'd be happy to do it.

I guess I could make a PR myself.

**Comment by RReverser** [comment]
> Also, updating the tests in this repo is pretty easy, just run update-tests.sh, and it makes the new commit for you. If that doesn't work for you I'd be happy to do it.

Looks like there are quite a few errors and in the end fails with:

```
>> git branch try-merge origin/master
fatal: Not a valid object name: 'origin/master'.
sub-command failed: git branch try-merge origin/master
```

It did change some files, but I'm not sure if it succeeded or failed halfway...

**Comment by aardappel** [comment]
I have only ever run the script, did not make it.
I am guessing it should be sufficient to replace occurrences of `master` with `main` ?

**Comment by RReverser** [comment]
Ah looks like author is @binji.

>  I am guessing it should be sufficient to replace occurrences of `master` with `main` ?

Maybe, but along the way it also printed a bunch of Git conflict messages and I'm not sure if those are normal part of the process or also require fixing.

**Comment by sbc100** [comment]
The conflict message happen for each repo that contains conflicts... the resulting change should still contains updates for any repos that don't contain conflicts and can be landed as is.

At the same time we normally report the conflicts upstream so that the upstream maintainers can fix the conflicts for the next update.

**Comment by RReverser** [comment]
Ok I'll give it another try, I guess partial update is better than nothing.

**Comment by RReverser** [comment]
> I am guessing it should be sufficient to replace occurrences of `master` with `main` ?

UPD: it's not so easy because some proposals like `simd` already switched to `main`, while others still use `master`, so this needs some extra logic to try both. Trying...

**Comment by RReverser** [comment]
Ugh, sorry, just noticed `update_repos` too... My shell-foo is not that good, I'll leave this to others who are more invested 😅 (alternatively, we could get other proposals and the spec to rename their branches from master -> main too for consistency which would simplify the logic again)

**Comment by sbc100** [comment]
Yes, lets file bugs with each repo that has not yet been updated.

**Comment by sbc100** [comment]
Actually it looks like the main spec repo (and indeed this repo) still use `master` so I guess it was `simd` that jumped the gun on that change.

**Pull request merged**
